src/Compilers/Shared/DesktopBuildClient.cs

```diff
@@ -455,6 +469,18 @@ private static bool CheckPipeConnectionOwnership(NamedPipeClientStream pipeStrea
             }
         }
 
+        private static ObjectSecurity GetPipeSecurity(PipeStream pipeStream) =>
```

Comment:
Why do we need reflection here?  In general the DesktopBuildClient.cs file should only be included when we know we're compiling for desktop.  Hence it should have the full 4.6 framework available.